proxy/client/services/lineage/service.go

```diff
@@ -21,17 +22,20 @@ type ILineageService interface {
 }
 
 func (s *Service) CreateLineage(lineageEvent string) int {
-	err := s.validator.Validate(lineageEvent)
-	if err != nil {
+	if err := s.validator.Validate(lineageEvent); err != nil {
 		go s.failedEventHandler.Handle(lineageEvent)
 		return http.StatusBadRequest
 	}
 
-	s.eventLogger.Log(lineageEvent)
+	if err := s.eventLogger.Log(lineageEvent); err != nil {
+		go s.failedEventHandler.Handle(lineageEvent)
+		return http.StatusInternalServerError
```

Comment:
Do we want to return a `500` to the caller? Or, can we return a more helpful message?